includes/admin/admin-actions.php

```diff
@@ -391,9 +391,143 @@ function _give_show_test_mode_notice_in_admin_bar( $wp_admin_bar ) {
 
 	return true;
 }
+
 add_action( 'admin_bar_menu', '_give_show_test_mode_notice_in_admin_bar', 1000, 1 );
 
 /**
+ * Add Link to Import page in from donation archive and donation single page
+ *
+ * @since 1.8.13
+ */
+function give_import_page_link_callback() {
+	?>
+    <a href="<?php echo esc_url( give_import_page_url() ); ?>"
+       class="page-import-action page-title-action"><?php esc_html_e( 'Import Donations', 'give' ); ?></a>
+
+    <style type="text/css">
+        <?php
+		// Check if view donation single page only.
+		if ( ! empty( $_REQUEST['view'] ) && 'view-payment-details' === (string) give_clean( $_REQUEST['view'] ) && 'give-payment-history' === give_clean( $_REQUEST['page'] ) ) {
+			?>
+        .wrap #transaction-details-heading {
+            display: inline-block;
+        }
+
+        <?php
+	} else {
+		?>
+        /* So the "New Donation" button aligns with the wp-admin h1 tag */
+        .wrap > h1 {
+            display: inline-block;
+            margin-right: 5px;
+        }
+
+        <?php
+	} ?>
+    </style>
+	<?php
+}
+
+
+add_action( 'give_payments_page_top', 'give_import_page_link_callback', 11 );
+add_action( 'give_view_order_details_before', 'give_import_page_link_callback', 11 );
+
+
+function give_donation_import_callback() {
+	$import_setting = array();
+	$fields         = isset( $_POST['fields'] ) ? $_POST['fields'] : null;
+
+	parse_str( $fields );
```

Comment:
@raftaar1191 use `give_clean` to sanitize values

Reply:
Done